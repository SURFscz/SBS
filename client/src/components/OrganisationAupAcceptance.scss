@import "../stylesheets/vars.scss";

div.mod-aup-organisation {
    margin-top: 20px;
    padding: 10px 0;
    border-top: 3px solid $lighter-grey;
    border-bottom: 3px solid $lighter-grey;

    div.organisation-section {
        display: flex;
        padding: 0 10px;
        align-items: center;
        width: 100%;
        min-height: 56px;
        margin-bottom: 15px;

        img {
            border-radius: 4px;
            max-width: 76px;
        }

        div.border-right {
            margin-right: 25px;
            padding-right: 25px;
            border-right: 1px solid $dark-grey;
        }

        table {
            width: 100%;

            td.logo {
                width: 85px;
            }

            td.organisation-detail {
                width: 90%;

                div.organisation {
                    display: flex;
                    flex-direction: column;
                    margin-left: 15px;

                    span.organisation-name {
                        margin-bottom: 6px;
                    }
                }

                div.policies {
                    display: flex;
                }

                div.contact {
                    display: flex;
                    align-content: center;

                    svg {
                        margin-right: 10px;
                    }
                }
            }

        }

        span.no-link {
            color: $darker-grey;
            font-style: italic;
            max-width: 220px;
            display: inline-block;
        }
    }

    p.aup-info {
        margin-bottom: 15px;
    }

    div.terms {
        margin: 25px 0 0 0;
    }


}