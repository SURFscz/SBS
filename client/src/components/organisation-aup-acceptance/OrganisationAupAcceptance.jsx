import React from "react";
import I18n from "../../locale/I18n";
import "./OrganisationAupAcceptance.scss";
import "../_welcome/welcome.scss";
import Logo from "../_redesign/Logo";
import {isEmpty} from "../../utils/Utils";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import CheckBox from "../checkbox/CheckBox";
import DOMPurify from "dompurify";

export default function OrganisationAupAcceptance({
                                                      adminEmails,
                                                      setDisabled,
                                                      disabled,
                                                      organisation
                                                  }) {
    const renderOrganisationAup = organisation => {
        const mails = encodeURI(adminEmails.join(","));
        return (
            <div className="organisation-section" key={organisation.id}>
                <table>
                    <thead/>
                    <tbody>
                    <tr>
                        <td className="logo">
                            {organisation.logo && <Logo src={organisation.logo} alt={organisation.name}/>}
                        </td>
                        <td className="organisation-detail">
                            <div className="organisation">
                                <span className="organisation-name">{organisation.name}</span>
                                <div className="policies">
                                    <div className="border-right">
                                        {organisation.accepted_user_policy ?
                                            <a href={organisation.accepted_user_policy}
                                               rel="noopener noreferrer"
                                               target="_blank">{I18n.t("service.accepted_user_policy")}</a> :
                                            <span className="no-link">{I18n.t("aup.service.noAup")}</span>
                                        }
                                    </div>
                                    <div className="contact">
                                        {!isEmpty(mails) && <a href={`mailto:${mails}`} rel="noopener noreferrer">
                                            <FontAwesomeIcon
                                                icon="envelope"/><span>{I18n.t("aup.service.contact")}</span>
                                        </a>}
                                    </div>
                                </div>
                            </div>
                        </td>
                    </tr>
                    </tbody>
                </table>
            </div>
        );
    }
    return (
        <div className="mod-aup-organisation">
            <h4 className="aup-services">
                {I18n.t("aup.organisation.title")}
            </h4>
            <p className="aup-info"
               dangerouslySetInnerHTML={{__html: DOMPurify.sanitize(I18n.t("aup.organisation.info", {name: organisation.name}))}}/>
            <p >
            </p>
            {renderOrganisationAup(organisation)}
            <div className="terms">
                <CheckBox name="aup-organisation"
                          value={!disabled}
                          info={I18n.t("aup.organisation.check")}
                          onChange={() => setDisabled(!disabled)}/>
            </div>
        </div>);

}
