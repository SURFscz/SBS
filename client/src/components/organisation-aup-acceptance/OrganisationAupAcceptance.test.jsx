import React from "react";
import {render} from "@testing-library/react";
import OrganisationAupAcceptance from "./OrganisationAupAcceptance";

jest.mock("@fortawesome/react-fontawesome", () => ({
    FontAwesomeIcon: () => <span>Icon</span>
}));

describe("OrganisationAupAcceptance", () => {
    it("matches snapshot with default render", () => {
        const {asFragment} = render(
            <OrganisationAupAcceptance
                adminEmails={["example@example.com"]}
                organisation={{name: "Test Organisation"}}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with custom props", () => {
        const {asFragment} = render(
            <OrganisationAupAcceptance
                organisationName="Test Organisation"
                aupUrl="https://example.com/aup"
                adminEmails={["example@example.com"]}
                organisation={{name: "Test Organisation"}}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
