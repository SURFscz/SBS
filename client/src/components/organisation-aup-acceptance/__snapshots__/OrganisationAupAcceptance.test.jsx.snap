// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganisationAupAcceptance matches snapshot with custom props 1`] = `
<DocumentFragment>
  <div
    class="mod-aup-organisation"
  >
    <h4
      class="aup-services"
    >
      Organisation AUP
    </h4>
    <p
      class="aup-info"
    >
      You are not a member of organisation 
      <strong>
        Test Organisation
      </strong>
      . Before you can continue you must review the AUP of this organisation. Confirm below whether you accept it.
    </p>
    <p />
    <div
      class="organisation-section"
    >
      <table>
        <thead />
        <tbody>
          <tr>
            <td
              class="logo"
            />
            <td
              class="organisation-detail"
            >
              <div
                class="organisation"
              >
                <span
                  class="organisation-name"
                >
                  Test Organisation
                </span>
                <div
                  class="policies"
                >
                  <div
                    class="border-right"
                  >
                    <span
                      class="no-link"
                    >
                      No AUP provided
                    </span>
                  </div>
                  <div
                    class="contact"
                  >
                    <a
                      href="mailto:example@example.com"
                      rel="noopener noreferrer"
                    >
                      <span>
                        Icon
                      </span>
                      <span>
                        Contact
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="terms"
    >
      <div
        class="sds--checkbox-container sds--tooltip-parent checkbox "
      >
        <input
          checked=""
          id="aup-organisation"
          type="checkbox"
        />
        <label
          for="aup-organisation"
        >
          <span
            class="sds--checkbox--visual"
          >
            <svg
              height="11.9"
              viewBox="0 0 13.6 11.9"
              width="13.6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m11.9.4-7 9.6-3.3-3.2c-.4-.4-1-.4-1.3 0-.4.4-.4 1 0 1.3l3.2 3.2q.6.6 1.5.6c.9 0 1.1-.3 1.4-.8l7.1-9.6c.3-.4.2-1-.2-1.3-.5-.3-1.1-.2-1.4.2"
              />
            </svg>
          </span>
          <span
            class="checkbox--text"
          >
            I agree to the organisation acceptable use policy
          </span>
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`OrganisationAupAcceptance matches snapshot with default render 1`] = `
<DocumentFragment>
  <div
    class="mod-aup-organisation"
  >
    <h4
      class="aup-services"
    >
      Organisation AUP
    </h4>
    <p
      class="aup-info"
    >
      You are not a member of organisation 
      <strong>
        Test Organisation
      </strong>
      . Before you can continue you must review the AUP of this organisation. Confirm below whether you accept it.
    </p>
    <p />
    <div
      class="organisation-section"
    >
      <table>
        <thead />
        <tbody>
          <tr>
            <td
              class="logo"
            />
            <td
              class="organisation-detail"
            >
              <div
                class="organisation"
              >
                <span
                  class="organisation-name"
                >
                  Test Organisation
                </span>
                <div
                  class="policies"
                >
                  <div
                    class="border-right"
                  >
                    <span
                      class="no-link"
                    >
                      No AUP provided
                    </span>
                  </div>
                  <div
                    class="contact"
                  >
                    <a
                      href="mailto:example@example.com"
                      rel="noopener noreferrer"
                    >
                      <span>
                        Icon
                      </span>
                      <span>
                        Contact
                      </span>
                    </a>
                  </div>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="terms"
    >
      <div
        class="sds--checkbox-container sds--tooltip-parent checkbox "
      >
        <input
          checked=""
          id="aup-organisation"
          type="checkbox"
        />
        <label
          for="aup-organisation"
        >
          <span
            class="sds--checkbox--visual"
          >
            <svg
              height="11.9"
              viewBox="0 0 13.6 11.9"
              width="13.6"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m11.9.4-7 9.6-3.3-3.2c-.4-.4-1-.4-1.3 0-.4.4-.4 1 0 1.3l3.2 3.2q.6.6 1.5.6c.9 0 1.1-.3 1.4-.8l7.1-9.6c.3-.4.2-1-.2-1.3-.5-.3-1.1-.2-1.4.2"
              />
            </svg>
          </span>
          <span
            class="checkbox--text"
          >
            I agree to the organisation acceptable use policy
          </span>
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
